GroundDBMemoryPool/src/rdma/rdma_client/write.cc

```diff
@@ -0,0 +1,32 @@
+#include <rdma.hh>
+#include "../util.h"
+
+int rdma_write(
+    struct resources *res, /* RDMA Connection resources */
+    const char *buffer,    /* Local buffer to write from */
+    const size_t size      /* number of bytes to write */
+)
+{
+    char backup[BUFFER_SIZE];
+    // Copy buffer to backup. Used for restoring buffer in case of failure
+    memcpy(backup, res->buf, BUFFER_SIZE);
```

Comment:
Unsure is it is needed.

Reply:
Maybe useful when multiple threads accessing `res->buf` simultaneously, but I cannot see how `backup` prevents it.